crates/aiken-lang/src/tipo/exhaustive.rs

```diff
@@ -575,11 +578,13 @@ pub(super) fn simplify(
                         module,
                         ..
                     } => (module, type_name, args.len()),
+                    tipo::Type::Pair { .. } => (empty, pair, 2),
                     _ => {
-                        unreachable!("ret should be a Type::App")
+                        unreachable!("ret should be a Type::App or Type::Pair")
                     }
                 },
-                _ => unreachable!("tipo should be a Type::App"),
+                tipo::Type::Pair { .. } => (empty, pair, 2),
+                _ => unreachable!("tipo should be a Type::App or Type::Pair"),
```

Comment:
That doesn't sound right but I suppose is an effect of how we added `Pair` as its own _type_ instead of making it an instance of an `App` type. `List` should probably follow the same path then.